### PR TITLE
Build Opus as shared library in static build, copy it into App Bundle

### LINKED ISSUE #3642: [macOS] Mumble fails to load Opus library in recent snapshots
After a clean system installation with Mojave 10.14.3 on my mac, I can't use the dev-version of Mumble 1.3. I can install and start the above mentioned version without any problems, but the speech with other users is not possible. I see that the other people are talking with the changing color of there icon, but I can hear nothing or just some strange noises like interfering noises. The same problem exists if I want to talk. I tried to change my audio setting, but I can't solve the problem.

The latest stable version of 1.2 is working without any problems.

/cc @baumannsven

### PR DESCRIPTION
**Pull request by davidebeatrici**
Fixes #3642.

We load Opus at runtime since #3431.

### CODE REVIEW

File: macx/scripts/osxdist.py [hunk 1]
```diff
@@ -129,12 +129,12 @@ def copy_resources(self, rsrcs):
...
 		'''
-			Try to copy the dynamic CELT libraries into the App Bundle.
+			Try to copy the dynamic audio codec libraries into the App Bundle.
 		'''
-		print ' * Attempting to copy CELT libraries into App Bundle'
+		print ' * Attempting to copy audio codec libraries into App Bundle'
 		dst = os.path.join(self.bundle, 'Contents', 'Codecs')
 		os.makedirs(dst)
-		codecs = ('release/libcelt0.0.7.0.dylib', 'release/libcelt0.0.11.0.dylib')
+		codecs = ('release/libcelt0.0.7.0.dylib', 'release/libcelt0.0.11.0.dylib', 'release/libopus.dylib')
```
Comment (hacst) [inline macx/scripts/osxdist.py]: There's a listing of binaries to sign further down in this file. Does libopus.dylib not also need to be listed there?

File: 3rdparty/opus-build/opus-build.pro [hunk 2]
```diff
@@ -20,6 +20,11 @@ VPATH = ../$$SOURCEDIR
 TARGET = opus
 DEFINES += HAVE_CONFIG_H
 
+CONFIG(static) {
```
Comment (hacst) [inline 3rdparty/opus-build/opus-build.pro]: Doesn't this also change behaviour for windows? How did we get the right binary there?

File: macx/scripts/osxdist.py (hunk 1 above)
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: Definitely, I forgot to add it.

File: 3rdparty/opus-build/opus-build.pro (hunk 2 above)
Comment (davidebeatrici) [reply 3rdparty/opus-build/opus-build.pro]: The library was already forced to be compiled as shared on Windows:
https://github.com/mumble-voip/mumble/blob/15072a4505af39f254fe0d35357d95b9a121e991/3rdparty/opus-build/opus-build.pro#L36-L37

Comment (davidebeatrici) [reply 3rdparty/opus-build/opus-build.pro]: I removed those lines, since they are redundant.

File: macx/scripts/osxdist.py [hunk 3]
```diff
@@ -296,7 +296,8 @@ def package_client():
 	a.copy_resources(['icons/mumble.icns'])
 	a.update_plist()
 	if not options.universal:
-		a.add_compat_warning()
+		if options.compat_warning:
+			a.add_compat_warning()
```
[the remark this answers is not in the document]
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: From the commit message:
> QMake always passes "-arch x86_64" after the parameters specified in our project file, meaning that we can't build the application as 32 bit.
>
> This option allows us to disable the feature (warning when trying to run Mumble on a 32 bit system) for the CI build.

File: 3rdparty/opus-build/opus-build.pro (hunk 2 above)
Comment (Kissaki) [reply 3rdparty/opus-build/opus-build.pro]: That means we also introduce it for Linux though!?

File: macx/scripts/osxdist.py [hunk 4]
```diff
@@ -129,12 +129,12 @@ def copy_resources(self, rsrcs):
 
 	def copy_codecs(self):
 		'''
-			Try to copy the dynamic CELT libraries into the App Bundle.
+			Try to copy the dynamic audio codec libraries into the App Bundle.
```
[the remark this answers is not in the document]
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: I agree.

File: macx/scripts/osxdist.py [hunk 5]
```diff
@@ -387,6 +388,7 @@ def package_server():
 	parser.add_option('', '--universal', dest='universal', help='Build an universal snapshot.', action='store_true', default=False)
 	parser.add_option('', '--only-appbundle', dest='only_appbundle', help='Only prepare the appbundle. Do not package.', action='store_true', default=False)
 	parser.add_option('', '--only-overlay', dest='only_overlay', help='Only create the overlay installer.', action='store_true', default=False)
+	parser.add_option('', '--no-compat-warning', dest='compat_warning', help='No warning message when running the image on x86.', action='store_false', default=True)
```
[the remark this answers is not in the document]
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: "This option should only be used when the warning application can't be built as 32 bit (rendering it useless)."

What do you think?

File: 3rdparty/opus-build/opus-build.pro (hunk 2 above)
Comment (davidebeatrici) [reply 3rdparty/opus-build/opus-build.pro]: That's correct.

File: macx/scripts/osxdist.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (hacst) [reply macx/scripts/osxdist.py]: Any idea why it suddenly would do that and not before?

[the remark this answers is not in the document]
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: We never built the installer on Travis CI before.

[the remark this answers is not in the document]
Comment (hacst) [reply macx/scripts/osxdist.py]: Why would it matter whether it's in travis though?

[the remark this answers is not in the document]
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: It shouldn't, indeed.

It's possible that `-arch -x86_64` is passed automatically in recent QMake versions.

[the remark this answers is not in the document]
Comment (hacst) [reply macx/scripts/osxdist.py]: Does it work in internal CI? Do we have a way to experiment with that? My assumption is that adding such a binary is important for the osx bundle experience or @mkrautz wouldn't have bothered to add it in the first place.

[the remark this answers is not in the document]
Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: Yes, it works in internal CI, `-arch x86_64` is not passed there.

The binary's purpose is to show an informative message in case the bundle is run on a 32 bit system:
https://github.com/mumble-voip/mumble/blob/15072a4505af39f254fe0d35357d95b9a121e991/macx/compat/compat.m#L21-L40

**Comment by mkrautz** [comment]
There's a reason we use static libraries. Because it makes code signing easier. We have no mechanism for signing shared libraries inside the App Bundle on macOS currently/anymore. So I think that should be fixed as well before we can merge this.

**Comment by mkrautz** [comment]
Disregard my comment above. I was obviously wrong, since osxdist.py does just that. :-)

File: macx/scripts/osxdist.py [hunk 6]
```diff
@@ -312,6 +312,7 @@ def package_client():
 			'release/Mumble.app/Contents/Plugins/libmanual.dylib',
 			'release/Mumble.app/Contents/Codecs/libcelt0.0.7.0.dylib',
 			'release/Mumble.app/Contents/Codecs/libcelt0.0.11.0.dylib',
+			'release/Mumble.app/Contents/Codecs/libopus.dylib',
```
Comment (mkrautz) [inline macx/scripts/osxdist.py]: This change is unrelated to this commit? Split it?

Comment (davidebeatrici) [reply macx/scripts/osxdist.py]: Thank you very much for noticing, this change should've been in b34ab4a4d4684874e0844db42921c2d82d30d45b.
